### PR TITLE
add build target to compile the starter if its present

### PR DESCRIPTION
**Pull request by dothebart**

### CODE REVIEW

File: scripts/build-bundle.sh [hunk 1]
```diff
@@ -12,6 +12,10 @@ for i in $@; do
     fi
 done
 
+if ! test -d 3rdParty/arangodb-starter; then
+    MOREOPTS="${MOREOPTS} --downloadStarter"
```
Comment (ObiWahn) [inline scripts/build-bundle.sh]: MOREOPTS+=( '--downloadStarter' )

File: scripts/build-bundle.sh [hunk 2]
```diff
@@ -12,6 +12,10 @@ for i in $@; do
     fi
 done
```
Comment (ObiWahn) [inline scripts/build-bundle.sh]: MOREOPTS=()

File: scripts/build-bundle.sh [hunk 3]
```diff
@@ -20,9 +24,9 @@ done
     --buildDir build-${EP}bundle \
     --prefix "/opt/arangodb" \
     --targetDir /var/tmp/ \
-    --downloadStarter \
     --clang \
     --staticOpenSSL \
+    ${MOREOPTS} \
```
Comment (ObiWahn) [inline scripts/build-bundle.sh]: "${MOEROPTS[@]}"

File: 3rdParty/CMakeLists.txt [hunk 4]
```diff
@@ -106,3 +106,32 @@ add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/snappy/google-snappy-d53de18)
 ################################################################################
```
Comment (ObiWahn) [inline 3rdParty/CMakeLists.txt]: This is ugly but it can not be helped:(

Copy-paste-leftovers like  ${NINJA_BUILD_DIR} could be removed.

**Comment by dothebart** [comment]
addresses https://github.com/arangodb/arangodb/issues/3604

**Pull request merged**